csrc/scheduler/pointwise.cpp

```diff
@@ -295,10 +295,17 @@ std::unique_ptr<PointwiseParams> getPointwiseHeuristics(
                 largest_out, true, true));
       });
 
+  int64_t max_dtype_size_for_vectorization = 1;
+  for (auto inp : vectorizable_inputs_outputs_entry.get()) {
+    max_dtype_size_for_vectorization = std::max(
+        max_dtype_size_for_vectorization,
+        (int64_t)dataTypeSize(inp->getDataType().value(), index_type));
+  }
+
   constexpr int64_t kSixteen = 16; // clang tidy
   auto max_vect_factor = ceilDiv(
       // Available vectorization based on size of data type
-      (int64_t)kSixteen / max_input_dtype_size,
+      (int64_t)kSixteen / max_dtype_size_for_vectorization,
```

Comment:
I think I should use this only for max_vect_factor?

I see `max_input_dtype_size` has also be used for unroll, so I should leave that as-is and not replace it with the max_dtype_size_for_vectorization.
cc'ing @liqiangxl

Reply:
I looked at some parts of the code, such as `getEmpiricalUnrollFactor`. It seems it shouldn't use the max size but it should use the size of each individual vectorized input.

Reply:
> I looked at some parts of the code, such as `getEmpiricalUnrollFactor`. It seems it shouldn't use the max size but it should use the size of each individual vectorized input.

You are right, all the vectorized inputs are using the same vectorization factor, may lead to different vectorized width.

Reply:
> I think I should use this only for max_vect_factor?
> 
> I see `max_input_dtype_size` has also be used for unroll, so I should leave that as-is and not replace it with the max_dtype_size_for_vectorization. cc'ing @liqiangxl

Does this influence the unroll factor in the targeted test? We can leave it here and revise when correcting the issues in `getEmpiricalUnrollFactor`